routers/web/repo/pull.go

```diff
@@ -318,8 +318,33 @@ func PrepareMergedViewPullInfo(ctx *context.Context, issue *models.Issue) *git.C
 	setMergeTarget(ctx, pull)
 	ctx.Data["HasMerged"] = true
 
+	var baseCommit string
+	// Some migrated PR won't have any Base SHA and lose history, try to get one
+	if pull.MergeBase == "" {
+		commitSHA := ctx.Repo.GitRepo.ReadPatch(pull.Index)
+		if commitSHA != "" {
+			// Get immediate parent of the first commit in the patch, grab history back
+			parentCommit, err := git.NewCommand("rev-list", "-1", "--skip=1", commitSHA).RunInDir(ctx.Repo.GitRepo.Path)
+			// Special case on Git < 2.25 that doesn't fail on immediate empty history
+			if err != nil || parentCommit == "" {
+				log.Info("No known parent commit for PR %d, error: %v", pull.Index, err)
+				// bring at least partial history if it can work
+				parentCommit = commitSHA
+			} else {
+				// Recreate pull head in files
+				err := ctx.Repo.GitRepo.WritePullHead(pull.Index, commitSHA)
+				if err != nil {
+					log.Error("Could not write head file", err)
```

Comment:
how to handle?

Reply:
I'm not sure if it's worth throwing errors there :
if you cannot read patch file or write head, you'll get back to "normal behavior" when the migrated repository has no history for this pull request, then does not show any files nor commits.

If there is an error at this point, then it must be that a problem happened with reading or writing on the disk, so it's a bigger problem than a simple "commit file".

But if you have any suggestion, I would gladly follow up :)